source/MaterialXCore/Override.h

```diff
@@ -0,0 +1,110 @@
+class MX_CORE_API Override
+{
+  public:
+    Override(
+        std::shared_ptr<Document> doc,
+        const vector<string>& properties,
+        const vector<ValuePtr>& values = {});
+    Override(const Override& other);
+    ~Override() { }
+
+    /// Create an override for this document with a list of properties.
+    /// @param doc The document to override.
+    /// @param properties Vector of property names, each one must match the path name of an input in the document.
+    /// @param values Optional vector of values to override the ones in the document, length must match properties.  If not provided values are initialized with defaults from the document.
+    /// @return Pointer to the new override object.
+    static OverridePtr create(
+        std::shared_ptr<Document> doc,
+        const vector<string>& properties,
+        const vector<ValuePtr>& values = {})
+    {
+        return std::make_shared<Override>(doc, properties, values);
+    }
+
+    /// Create override from an existing override object.
+    /// @param other An existing Override to copy.
+    static OverridePtr copy(OverridePtr other)
+    {
+        return std::make_shared<Override>(*other);
+    }
+
+    /// Set the value of the named override property.
+    /// @param name The name of the property to set.
+    /// @param val The value to set, any valid MaterialX type.
+    template <typename T>
+    void setValue(const string& name, const T& val)
+    {
+        int idx = getIndex(name);
+        _values[idx] = Value::createValue(val);
+    }
+
+    /// Get the index of the named property
+    /// @param name The name of the property to lookup.
+    /// @return The index (or -1 if property name is not found.)
+    int getIndex(const string& propName)
+    {
+        auto iter = _indexLookup.find(propName);
+        if (iter == _indexLookup.end())
+            return -1;
+
+        return iter->second;
+    }
+
+    /// Get the document for this override.
+    /// @return The document being overriden.
+    std::shared_ptr<Document> getDocument() { return _doc; }
+
+    /// Get the document input for the Nth property being overriden.
+    /// @return Pointer to the input within the document.
+    std::shared_ptr<Input> getPropertyInput(int n) { return _propertyInputs[n]; }
+
+    /// Get the current value of of the Nth property being overriden.
+    /// @return The current value.
+    ValuePtr getValue(int n) { return _values[n]; }
+
+    /// Get the name of the Nth property being overriden.
+    /// @return The full path name of the input associated with the override property.
+    string getPropertyName(int n) { return _properties[n]; }
+
+    /// Get the number of properties in this override.
+    /// @return The property count.
+    int getPropertyCount() { return int(_properties.size()); }
+
+    // TODO: Add the ability to serialize overrides (to/from XML, JSON, etc.)
```

Comment:
As mentioned we should have serialize functions as with regular materialX documents, and the ability to produce a document from an override (for implementations that do not want to implement a different data path for override materials)